content_scripts/utils.js

```diff
@@ -200,18 +200,19 @@ function actionWithSelectionPreserved(cb) {
 }
 
 function filterAncestors(elements) {
-    var tmp = [];
-    if (elements.length > 0) {
-        // filter out element which has its children covered
-        tmp = [elements[elements.length - 1]];
-        for (var i = elements.length - 2; i >= 0; i--) {
-            if (!elements[i].contains(tmp[0])) {
-                tmp.unshift(elements[i]);
-            }
+    if (elements.length === 0) {
+        return elements;
+    }
+
+    // filter out element which has its children covered
+    let result = [elements.pop()];
+    for (let i = elements.length - 1; i >= 0; i--) {
+        if (!elements[i].contains(result[0])) {
```

Comment:
Is it possible to check `runtime.conf.clickableSelector` here? So that we could avoid anther loop through `elements`. Performance of listing elements really matters.

Reply:
Thanks for the feedback! I'll take a look at it. 

In the spirit of making this faster - I was thinking about replacing `unshift` with `push` as it's considerably faster (https://stackoverflow.com/questions/12250697/time-complexity-of-unshift-vs-push-in-javascript).

But I'm not quite sure I understand the logic behind this loop. 
Right now  we take the last element, put it into result then look at the previous element and check whether it contains the last element we've put to the result. 
This seems to imply that the original list of elements we get is ordered as:
`parent 1 > parent 0> child`. 

Is that indeed a guarantee we rely on here?

Reply:
Yes, current logic relies on this kind of assumption as it behaves now. Using `unshift` here is just to keep the order, so that user would be confused by the Hint labels. Improvement is also welcomed. Thanks!

Reply:
I see. pushed an update that migrates to using `push` but still preserves the original order